pepper-apis/src/main/java/org/broadinstitute/ddp/route/PutFormAnswersRoute.java

```diff
@@ -242,29 +248,43 @@ private FormInstance loadFormInstance(Response response,
         return form;
     }
 
+    // For a parent instance, check if there are any child instances that are not hidden and if those are complete.
     private void checkChildInstancesCompleteness(Response response,
                                                  Handle handle,
                                                  String userGuid,
                                                  String operatorGuid,
                                                  String instanceGuid,
+                                                 FormInstance form,
                                                  LanguageDto preferredLangDto,
                                                  UserActivityInstanceSummary instanceSummary) {
-        // For a parent instance, check if there are any child instances and if those are complete.
-        List<ActivityInstanceDto> childInstances = instanceSummary.getInstancesStream()
+        Map<String, List<ActivityInstanceDto>> childInstances = new HashMap<>();
+        instanceSummary.getInstancesStream()
                 .filter(instance -> instanceGuid.equals(instance.getParentInstanceGuid()))
-                .collect(Collectors.toList());
-        for (var childInstanceDto : childInstances) {
-            if (childInstanceDto.getStatusType() != InstanceStatusType.COMPLETE) {
-                // Child instance is not finished but it might not have required questions, so check it.
-                FormInstance childForm = loadFormInstance(
-                        response, handle, userGuid, operatorGuid,
-                        childInstanceDto.getGuid(), preferredLangDto, instanceSummary);
-                if (!childForm.isComplete()) {
-                    String msg = "Status for instance " + instanceGuid + " cannot be set to COMPLETE because the"
-                            + " question requirements are not met for child instance " + childInstanceDto.getGuid();
-                    LOG.info(msg);
-                    throw ResponseUtil.haltError(response, 422,
-                            new ApiError(ErrorCodes.QUESTION_REQUIREMENTS_NOT_MET, msg));
+                .forEach(instance -> childInstances
+                        .computeIfAbsent(instance.getActivityCode(), key -> new ArrayList<>())
+                        .add(instance));
+        for (FormSection section : form.getAllSections()) {
+            for (FormBlock block : section.getBlocks()) {
+                if (block.getBlockType() != BlockType.ACTIVITY || !block.isShown()) {
```

Comment:
Does it make sense to implement a method `blockIsActivityAndShown()` and apply it like this:
```
for (...) {
  if (blockIsActivityAndShown(block)) {
  ...
  }
}
```

Reply:
It's just two conditions in the `if` so I think we can leave it for now.